services/m3aggregator/config/aggregator.go

```diff
@@ -0,0 +1,353 @@
+	"github.com/m3db/m3x/pool"
+)
+
+var (
+	errUnknownQuantileSuffixFnType = errors.New("unknown quantile suffix function type")
+	errUnknownFlushFnType          = errors.New("unknown flush function type")
+	errNoForwardFlushConfiguration = errors.New("no forward flush configuration")
+	emptyPolicy                    policy.Policy
+)
+
+// aggregatorConfiguration contains aggregator configuration.
+type aggregatorConfiguration struct {
+	// Common metric prefix.
+	MetricPrefix string `yaml:"metricPrefix"`
+
+	// Counter metric prefix.
+	CounterPrefix string `yaml:"counterPrefix"`
+
+	// Timer metric prefix.
+	TimerPrefix string `yaml:"timerPrefix"`
+
+	// Timer sum metric suffix.
+	TimerSumSuffix string `yaml:"timerSumSuffix"`
+
+	// Timer sum square metric suffix.
+	TimerSumSqSuffix string `yaml:"timerSumSqSuffix"`
+
+	// Timer sum mean metric suffix.
+	TimerMeanSuffix string `yaml:"timerMeanSuffix"`
+
+	// Timer lower metric suffix.
+	TimerLowerSuffix string `yaml:"timerLowerSuffix"`
+
+	// Timer upper metric suffix.
+	TimerUpperSuffix string `yaml:"timerUpperSuffix"`
+
+	// Timer count metric suffix.
+	TimerCountSuffix string `yaml:"timerCountSuffix"`
+
+	// Timer standard deviation metric suffix.
+	TimerStdevSuffix string `yaml:"timerStdevSuffix"`
+
+	// Timer median metric suffix.
+	TimerMedianSuffix string `yaml:"timerMedianSuffix"`
+
+	// Timer quantile suffix function type.
+	TimerQuantileSuffixFnType string `yaml:"timerQuantileSuffixFnType"`
+
+	// Gauge metric suffix.
+	GaugePrefix string `yaml:"gaugePrefix"`
+
+	// Stream configuration for computing quantiles.
+	Stream streamConfiguration `yaml:"stream"`
+
+	// Minimum flush interval across all resolutions.
+	MinFlushInterval time.Duration `yaml:"minFlushInterval"`
+
+	// Maximum flush size in bytes.
+	MaxFlushSize int `yaml:"maxFlushSize"`
+
+	// Flushing configuration.
+	Flush flushConfiguration `yaml:"flush"`
+
+	// EntryTTL determines how long an entry remains alive before it may be expired due to inactivity.
+	EntryTTL time.Duration `yaml:"entryTTL"`
+
+	// EntryCheckInterval determines how often entries are checked for expiration.
+	EntryCheckInterval time.Duration `yaml:"entryCheckInterval"`
+
+	// EntryCheckBatchPercent determines the percentage of entries checked in a batch.
+	EntryCheckBatchPercent float64 `yaml:"entryCheckBatchPercent" validate:"min=0.0,max=1.0"`
+
+	// Pool of entries.
+	EntryPool pool.ObjectPoolConfiguration `yaml:"entryPool"`
+
+	// Pool of counter elements.
+	CounterElemPool pool.ObjectPoolConfiguration `yaml:"counterElemPool"`
+
+	// Pool of timer elements.
+	TimerElemPool pool.ObjectPoolConfiguration `yaml:"timerElemPool"`
+
+	// Pool of gauge elements.
+	GaugeElemPool pool.ObjectPoolConfiguration `yaml:"gaugeElemPool"`
+
+	// Pool of buffered encoders.
+	BufferedEncoderPool pool.ObjectPoolConfiguration `yaml:"bufferedEncoderPool"`
+}
+
+// NewAggregatorOptions creates a new aggregator options.
+func (c *aggregatorConfiguration) NewAggregatorOptions(
```

Comment:
if aggregatorConfiguration is already private, should NewAggregatorOptions be private too?